blog: ddev-upsun add-on blog (#440)
ddev/ddev.com: src/content/blog/ddev-upsun-platformsh-addon.md

---
title: "Upsun Support for DDEV in a New Add-on"
pubDate: 2025-10-02
#modifiedDate: 2025-04-07
summary: "New DDEV Add-on supports Upsun Flex and Upsun Fixed (Platform.sh-style configuration)"
author: Randy Fay
featureImage:
  src: /logos/upsun.svg
  alt: Sophisticated upsun support for DDEV
categories:
  - Announcements
  - Guides
---

## Introduction

We're proud to announce that DDEV now has an [Upsun add-on](https://github.com/ddev/ddev-upsun) with extensive support for Upsun Flex and Upsun Fixed (Platform.sh-style configuration)!

> Although DDEV has had `ddev pull upsun` support for Upsun projects for a couple of years, the add-on provides a more complete integration with Upsun. It reads your Upsun configuration and automatically configures matching services in your local DDEV project, mirroring what your upstream project uses.

Reviewer: ```suggestion
Although DDEV has had `ddev pull upsun` support for Upsun projects for a couple of years, the add-on provides a more complete integration. It reads your Upsun configuration and automatically configures matching services in your local DDEV project, mirroring what your upstream project uses.
```
the first sentence had the term upsun three times and the second sentence starts of with another mention of the name. i wonder if dropping the mention at least once might be an option?

Although many Upsun-hosted sites are basic nginx-fpm-database sites that DDEV has easily supported for years, the more complex ones have required special handling, like having to explicitly set PHP version, database type, etc. Now the add-on can do much of that for you.

For example, if your Upsun project uses PostgreSQL 15 with Redis and a specific PHP version, previously you'd need to manually configure each in your `.ddev/config.yaml`. Now, the add-on detects and configures all of this automatically.

Upsun is a great platform for developers, and so many of you use it in the new "Flex" style and the previous "Fixed" style (Platform.sh-style). The add-on makes it easy to use both with a single add-on.

And of course Upsun is the lead sponsor of DDEV, so we always delight in making it work for you.

---

## The Future of `ddev-upsun`

A sophisticated platform like Upsun has many possibilities, and we want most things to work on most projects. The scope will be limited to PHP for now. However, we want to add:

- Versioning for add-ons (Redis version will be supported in `ddev/ddev-redis` add-on, for example).
- More service support, like Elasticsearch and Solr.

> - Even more extensive automated tests. Currently we have a very extensive set of tests based on Drupal setup, but want to add tests for other frameworks and CMSes.

Reviewer: ```suggestion
- Even more extensive automated tests. Currently we have a very extensive set of tests based on a Drupal setup, but want to add tests for other frameworks and CMSes.
```
does that need an article? sounds slightly incomplete?

---

## How We Got Here (PHP Add-ons)

> DDEV v1.24.8 (required) adds [support for add-ons written mostly in PHP](https://docs.ddev.com/en/stable/users/extend/creating-add-ons/#php-based-actions-new). Originally, all add-ons were written in Bash, with Go templating. It worked great for so many simple add-ons, but was way too complicated for sophisticated add-ons. PHP is a familiar and powerful language for many add-on developers, so now provides a much more expressive way to write complex add-ons.

Reviewer: ```suggestion
DDEV v1.24.8 (required) adds [support for add-ons written mostly in PHP](https://docs.ddev.com/en/stable/users/extend/creating-add-ons/#php-based-actions-new). Originally, all add-ons were written in Bash, with Go templating. It worked great for so many simple add-ons, but was way too complicated for sophisticated ones. PHP is a familiar and powerful language for many add-on developers, so now provides a much more expressive way to write complex add-ons.
```

the "so now provides" at the end of the last sentence might be missing a reference? "so" throws me off somehow. would using something like "it" be better? something like "..., it provides a much more expressive way to write complex add-ons now." ?

The [ddev/platformsh](https://github.com/ddev/platformsh) add-on forced the issue on this. Every time we went back to it to maintain it we had to remember how Go templates worked and sort out complex Bash logic. Bash and Go templates are a lousy combination for maintainability. PHP should be much better. The `ddev-upsun` add-on will replace `ddev/platformsh` and supports both the Flex and Fixed Upsun configurations.

If you're currently using the `ddev/ddev-platformsh` add-on, you can migrate to `ddev-upsun` by removing the old add-on with `ddev add-on remove ddev-platformsh` and adding the new one with `ddev add-on get ddev/ddev-upsun`.

---

## What's Next

- Check out the [full documentation](https://github.com/ddev/ddev-upsun#readme) for advanced configuration options.

> - We'd love feedback—open issues or contribute PRs if you encounter edge cases.

Reviewer: ```suggestion
- We'd love your feedback — open issues or contribute PRs if you encounter any edge case.
```
just added spaces before and after the dash. with them it read like one thing aka "feedback-open" and add too more minor changes.
